### PR TITLE
NOTICK Remove unneeded inheritance

### PR DESCRIPTION
**Pull request by kyriathar**
- drops crypto request handler API and removes unneeded inheritance.

### CODE REVIEW

**Review by driessamyn — DISMISSED** [review]
LGTM

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ * Handles crypto requests and returns response. Because requests are coming in as (Avro) [Object]s we grab appropriate handler
+ * by actual request's [Class].
+ */
+interface CryptoRequestHandler<REQUEST, RESPONSE> {
+    val requestClass: Class<REQUEST>
+    fun handle(request: REQUEST, context: CryptoRequestContext): RESPONSE
+}
+
+fun Map<Class<*>, CryptoRequestHandler<*, out Any>>.getHandlerForRequest(requestType: Class<*>): CryptoRequestHandler<Any, out Any> =
+    uncheckedCast(
```
Comment (dickon) [inline components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt]: I think I'd like to find a way to do this that avoids the unchecked cast.

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt [hunk 2]
```diff
@@ -0,0 +1,19 @@
+package net.corda.crypto.service.impl
+
+import net.corda.data.crypto.wire.CryptoRequestContext
+import net.corda.v5.base.util.uncheckedCast
+
+/**
+ * Handles crypto requests and returns response. Because requests are coming in as (Avro) [Object]s we grab appropriate handler
+ * by actual request's [Class].
+ */
+interface CryptoRequestHandler<REQUEST, RESPONSE> {
```
Comment (dickon) [inline components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt]: So, I happened to look at this and wrote up https://github.com/corda/platform-eng-design/blob/d404f9a9dbc58538cbfb71d3d81f7486bd4c859d/core/corda-5/corda-5.0/cipher-suite/abstractions-and-tech-debt.md  last section on WireProcessor.

I believe the right thing to do is to remove the whole `WireProcessor` abstraction, which effectively you've renamed here to `CryptoRequestHandler`, all the handler inner classes, and just have a big `when` block in the `onNext` handle.

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/bus/CryptoFlowOpsBusProcessor.kt [hunk 3]
```diff
@@ -139,8 +148,10 @@ class CryptoFlowOpsBusProcessor(
 
     private class SignFlowCommandHandler(
         private val client: CryptoOpsProxyClient
-    ) : Handler<SignFlowCommand> {
-        override fun handle(context: CryptoRequestContext, request: SignFlowCommand): Any =
+    ) : CryptoRequestHandler<SignFlowCommand, CryptoSignatureWithKey> {
+        override val requestClass = SignFlowCommand::class.java
+
+        override fun handle(request: SignFlowCommand, context: CryptoRequestContext): CryptoSignatureWithKey =
```
Comment (dickon) [inline components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/bus/CryptoFlowOpsBusProcessor.kt]: So the problem I have with this pattern is that it's even more lines of code than the current version, whereas I'd like to get smaller.

**Review by dickon — CHANGES_REQUESTED** [review]
Let's just kill the whole `WireProcessor` abstraction.

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt (hunk 2 above)
Comment (kyriathar) [reply components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt]: Done

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/bus/CryptoFlowOpsBusProcessor.kt (hunk 3 above)
Comment (kyriathar) [reply components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/bus/CryptoFlowOpsBusProcessor.kt]: Removed

File: components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt (hunk 1 above)
Comment (kyriathar) [reply components/crypto/crypto-service-impl/src/main/kotlin/net/corda/crypto/service/impl/CryptoRequestHandler.kt]: Unchecked casts removed

**Review by dickon — APPROVED** [review]
Lovely work, much cleaner now.

**Pull request merged**